app/assets/javascripts/controllers/floating_ip/floating_ip_form_controller.js

```diff
@@ -42,34 +50,18 @@ ManageIQ.angular.app.controller('floatingIpFormController', ['$http', '$scope',
 
   vm.resetClicked = function() {
     vm.floatingIpModel = angular.copy( vm.modelCopy );
-    $scope.angularForm.$setPristine(true);
+    vm.angularForm.$setPristine(true);
     miqService.miqFlash("warn", "All changes have been reset");
   };
 
   vm.filterNetworkManagerChanged = function(id) {
-    miqService.sparkleOn();
-    $http.get('/floating_ip/networks_by_ems/' + id)
-      .then(getNetworkByEmsFormData)
-      .catch(miqService.handleFailure);
-
+    miqService.getCloudNetworksByEms(function(data) {
```

Comment:
Why did you move API call to `miqService`? If it's not something common for multiple controllers it should stay in its own controller.

Reply:
@ZitaNemeckova, I'm not following you, the `getCloudNetworksByEms` which is now in `miq_service.js` is using both a Promise [1], the `then` part, where the promise is the callback itself.

[1] https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/assets/javascripts/miq_api.js#L5

Reply:
Yeah, I suggested to use Promise instead of callback but that's out of the scope for this PR. And this only populates fields after a change there's no need for returning Promise. Please ignore that comment :)